src/middlewares/authorize.js

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable import/prefer-default-export */
+import dotenv from 'dotenv';
+import jwt from 'jsonwebtoken';
+import ErrorResponse from '../utils/errorResponse';
+import {
+  isAdmin,
+  isSuperAdmin as superUser,
+  isManager,
+} from '../services/rolesServices';
+import { Users } from '../database/models';
+
+dotenv.config();
+const jwtToken = process.env.TOKEN_SECRET;
+const isSuperAdmin = (req, res, next) => {
+  if (req.headers.authorization) {
+    const token = req.headers.authorization.split(' ')[1];
+    if (!token) {
+      ErrorResponse.forbiddenError('Unauthenticated', res);
+    }
+    jwt.verify(token, jwtToken, (err, decoded) => {
+      if (err) {
+        ErrorResponse.forbiddenError('Unauthenticated', res);
+      } else {
+        const { email } = decoded;
+        const { id } = decoded;
+        if (email != null && id != null) {
+          const valid = { email, id };
+          req.valid = valid;
```

Comment:
this should already be a separate middleware for verifying the token. Why are you calling it "isSuperAdmin"